src/components/Api.js

```diff
@@ -0,0 +1,87 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(`${url}${options}`, { headers: this._headers }).then(
+      this._checkPromise
+    );
+  }
+
+  _checkPromise(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return this._request(this._baseURL, "/cards");
+  }
+
+  getUserInfo() {
+    return this._request(this._baseURL, "/users/me").then((userData) => {
+      return userData;
+    });
+  }
+
+  updateUserInfo(name, about) {
+    return fetch(`${this._baseURL}/users/me`, {
```

Comment:
**Could be improved**

`this._request` can be used everywhere


```js

  updateUserInfo(name, about) {
    return this._request(`${this._baseURL}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name,
        about,
      }),
    });
  }

```

Reply:
How can this._request be used everywhere if the different methods have different method properties and body properties?

Reply:
oh nevermind, i see answer below...